### PR TITLE
Feature/migrating to use gradle

### PR DESCRIPTION
**Pull request by rcordeau**
Removed subprojects and their build.gradle files in favor of dynamic task generation

---
This change is [Reviewable](https://reviewable.io/reviews/smarterapp/rdw_schema/50)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,22 +11,33 @@ plugins {
...
+               clean: org.flywaydb.gradle.task.FlywayCleanTask,
+               info: org.flywaydb.gradle.task.FlywayInfoTask,
+               validate: org.flywaydb.gradle.task.FlywayValidateTask,
+               baseline: org.flywaydb.gradle.task.FlywayBaselineTask,
+               repair: org.flywaydb.gradle.task.FlywayRepairTask]
+
+//Gradle can't call a single task more than once, because of its design, so
+//we dynamically create individual tasks for each of the available flyway commands
+//on all of our databases (dev and test)
+["reporting", "reporting-test", "warehouse", "warehouse-test"].each { schema ->
```
Comment (rcordeau) [inline build.gradle]: The main driver behind all of this change was the introduction of test databases for integration testing. I didn't want to have duplicated code all over the place, nor duplicated properties, so dynamic task generation was in order.

File: reporting/sql/V201702061486427077__initial_ddl.sql [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+** 	Initial script for the SBAC Reporting Data Warehouse schema
 **
 **  NOTES
 **  This schema assumes the following:
 **     1. one state (aka tenant) per data warehouse
 **     2. not all data elements from TRT are included, only those that are required for the current reporting
 **/
 
-ALTER DATABASE reporting CHARACTER SET utf8 COLLATE utf8_unicode_ci;
+ALTER DATABASE `${schemaName}` CHARACTER SET utf8 COLLATE utf8_unicode_ci;
```
Comment (rcordeau) [inline reporting/sql/V201702061486427077__initial_ddl.sql]: The schemaName is filled in by Flyway via the placeholders property.

File: README.md [hunk 3]
```diff
@@ -1,47 +1,54 @@
...
 To install or migrate, run:
 ```bash
-RDW_Schema$ ./gradlew migrateWarehouse
+RDW_Schema$ ./gradlew migrateWarehouse (or migrateWarehouse-test)
 OR
-RDW_Schema$ ./gradlew migrateReporting
+RDW_Schema$ ./gradlew migrateReporting (or migrateReporting-test)
 OR
-RDW_Schema$ ./gradlew migrateAll
+RDW_Schema$ ./gradlew migrateAll (migrates the dev and test instances for the schemas)
```
Comment (malaffoon) [inline README.md]: I wonder if we want to separate `migrateAll` and `migrateAll-test`

Comment (rcordeau) [inline README.md]: Yeah we can break out the test and non-test. Maybe keep the "All" as really "all", tests included.

**Pull request merged**